model_table_has_many.go

```diff
@@ -172,5 +172,18 @@ func indirectAsKey(field reflect.Value) interface{} {
 	if field.IsNil() {
 		return nil
 	}
+
+	if valuer, ok := field.Interface().(driver.Valuer); ok {
+		if v, err := valuer.Value(); err == nil {
+			switch reflect.TypeOf(v).Kind() {
+			case reflect.Array, reflect.Chan, reflect.Func,
+				reflect.Map, reflect.Ptr, reflect.Slice, reflect.UnsafePointer:
+				break
+			default:
+				return v
+			}
+		}
```

Comment:
I think we should always use the value returned by `driver.Valuer` even if it is not comparable.

I suggest to do this:

```
val, err := valuer.Value()
if err != nil {
    slog.Error(...)
}
return val // might be nil
```

Reply:
Maybe we don't need to log because the panic will come later

Reply:
It might panic on nil value (or not), but you won't see the error message in app logs which will make debugging slightly more complicated... Up to you.